File: src/main/kotlin/dev/jefy/connectpro/management/appliacaion/query/BadgeQuery.java
package dev.jefy.connectpro.management.appliacaion.query;

import org.jspecify.annotations.NullMarked;

import java.util.List;

import dev.jefy.connectpro.management.appliacaion.dtos.BadgeResponse;
import dev.jefy.connectpro.management.domain.vo.BadgeId;

/**
 * @author Jôph Yamba
 */
@NullMarked
public interface BadgeQuery {
    BadgeResponse get(BadgeId badgeId);
    List<BadgeResponse> getAll();
}

File: src/main/kotlin/dev/jefy/connectpro/management/appliacaion/query/BadgeQuery.kt
package dev.jefy.connectpro.management.appliacaion.query

import dev.jefy.connectpro.management.appliacaion.dtos.BadgeResponse
import dev.jefy.connectpro.management.domain.vo.BadgeId
import org.jspecify.annotations.NullMarked

/**
 * @author Jôph Yamba
 */

interface BadgeQuery {
    fun get(badgeId: BadgeId): BadgeResponse
    fun gatAll() : List<BadgeResponse>
}
